src/librustc/infer/mod.rs

```diff
@@ -1702,17 +1679,7 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
         }
 
         let closure_ty = self.tcx.closure_type(def_id, substs);
-        if self.normalize {
-            let closure_ty = self.tcx.erase_regions(&closure_ty);
-
-            if !closure_ty.has_projection_types() {
-                return closure_ty;
-            }
-
-            self.normalize_projections_in(&closure_ty)
-        } else {
-            closure_ty
-        }
+        closure_ty
```

Comment:
It only exists because it's doing something different than `self.tcx.closure_type(def_id, substs)`, but *not* after your changes. ~~The `InferTables` enum might also be removable~~ Nevermind, that's mainly there for lifetime issues.